CHANGES.rst

```diff
@@ -403,6 +403,9 @@ astropy.stats
 astropy.table
 ^^^^^^^^^^^^^
 
+- Fixed a bug when a column in ``QTable`` is converted to ``Quantity``,
+  datatype is sometimes not preserved. [#10950]
```

Comment:
I think you can be more specific here. At least from my thinking about it, the only case of dtype not being preserved is when the original column is an integer type. So you could say:
```
Fixed a bug when an integer-type ``Column`` in ``QTable`` is converted to ``Quantity``
by setting the unit, the datatype was changed to float. [#10950]
```

Reply:
Feedback incorporated.

BTW, do you happen to know the reason(s) why `QTable` performs `Quantity` conversion only for columns with units, rather than all numeric columns? As it is, users would need to deal with 2 different APIs (`Quantity` and `Column`) when using `QTable` to handle with a mix of numeric data with and without unit, e.g., flux, cadence, quality flags.